tests/NexusMods.MnemonicDB.Tests/DbTests.cs

```diff
@@ -675,6 +675,56 @@ public async Task ObserveLargeDatomChanges()
         allLoadouts.Should().Be(10000);
     }
 
+    [Fact]
+    public async Task CanNestObserveDatoms()
+    {
+        // To test nesting of observables, we're going to observe all loadouts, and then inside that observe all mods
+
+        List<EntityId> modIds = new();
+
+        using var disposable = Connection.ObserveDatoms(Loadout.Name)
+            .QueryWhenChanged(loadouts =>
+            {
+                foreach (var loadout in loadouts.Items)
+                {
+                    var mods = Connection.ObserveDatoms(Mod.Loadout, loadout.E)
+                        .QueryWhenChanged(mods =>
+                        {
+                            modIds.AddRange(mods.Items.Select(m => m.E));
+                            return 0;
+                        })
+                        .Subscribe();
+                }
+                return 42;
+            })
+            .Subscribe();
+        
+        using var tx = Connection.BeginTransaction();
+        var loadout = new Loadout.New(tx)
+        {
+            Name = "Test Loadout"
+        };
+        
+        for (var i = 0; i < 10; i++)
+        {
+            _ = new Mod.New(tx)
+            {
+                Name = $"Test Mod {i}",
+                Source = new Uri("http://test.com"),
+                LoadoutId = loadout
+            };
+        }
+        
+        await tx.Commit();
+        
+        // Delay because the chain of observables is async
+        await Task.Delay(1000);
```

Comment:
The fact that this is needed breaks a lot of our code, our code so far has always assumed that when `await tx.Commit();`  returned, all observables would have already been notified. 

Using Create async in `ObserveDatoms()` makes it so that some subscribers might get notified during the commit, some after the commit. 

I think knowing that all the subscribers have been notified after Commit returns is very valuable and we should not discard that assumption.

Reply:
We're not discarding this behavior. The only reason the delay is required here is due to the nested observables, otherwise all our other existing tests that assume instant subscription continue to work as expect. I realize there may be some lag here for nested observables, I'll monitor this and see if it cauees issues in the app code. The next solution for this (if this one doesn't work) is fairly complex so I'm trying to avoid having to implement that method.

Really the issue here is that Rx is a mess of locks without any clear definition of how those locks should be resolved.